src/App.tsx

```diff
@@ -17,18 +77,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchQuery={searchQuery}
+                setSearchQuery={setSearchQuery}
+                selectedStatus={selectedStatus}
+                setSelectedStatus={setSelectedStatus}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {hasErrorFromServer
+                ? <p>Error while fetching</p>
+                : (
+                  <TodoList
+                    todos={visibleTodos}
+                    setSelectedTodoId={setSelectedTodoId}
```

Comment:
don't pass function for updating state in child component, create wrapper for it, fix in all places